backend/src/api/datasource/datasource.controller.spec.ts

```diff
@@ -0,0 +1,160 @@
+import { Test } from '@nestjs/testing';
+import { PrismaModule } from '../../db/prisma.module';
+import { FitnessService } from '../../integration/fitness/fitness.service';
+import { DeepMockProxy, mockDeep } from 'jest-mock-extended';
+import { MockProvider } from '../../integration/fitness/providers/mock.provider';
+import { DatasourceController } from './datasource.controller';
+import { NestRequest } from '../../types/request.type';
+import { FitnessRepository } from '../../db/repositories/fitness.repository';
+import { Response } from 'express';
+import { TestConstants } from '../../../test/lib/constants';
+import { ConsoleLogger } from '@nestjs/common';
+import { LOGGER_SERVICE } from '../../logger/logger.service';
+
+describe('Datasource controller', () => {
+  let fitnessService: DeepMockProxy<FitnessService>;
+  let fitnessController: DeepMockProxy<DatasourceController>;
+  let fitnessRepository: DeepMockProxy<FitnessRepository>;
+
+  beforeAll(async () => {
+    const module = await Test.createTestingModule({
+      imports: [PrismaModule],
+      providers: [
+        FitnessService,
+        { useClass: ConsoleLogger, provide: LOGGER_SERVICE },
+      ],
+      controllers: [DatasourceController],
+    })
+      .overrideProvider(FitnessService)
+      .useValue(mockDeep<FitnessService>())
+      .overrideProvider(FitnessRepository)
+      .useValue(mockDeep<FitnessRepository>())
+      .compile();
+
+    fitnessService = module.get(FitnessService);
+    fitnessController = module.get(DatasourceController);
+    fitnessRepository = module.get(FitnessRepository);
+  });
+
+  it('It should return the datasources for a user', async () => {
+    // Mock the datasources
+    fitnessService.getDatasourcesForUser.mockResolvedValue([
+      new MockProvider(),
+    ]);
+
+    const mockRequest = {
+      user: {
+        id: 'MOCK',
+      },
+    } as NestRequest;
+
+    const userProviders =
+      await fitnessController.getDatasourcesForUser(mockRequest);
+
+    expect(userProviders.length).toBe(1);
+  });
+
+  it('Should return a specific datasource for the user', async () => {
+    fitnessService.getProviderForUserById.mockResolvedValue(new MockProvider());
+
+    const mockRequest = {
+      user: {
+        id: 'MOCK',
+      },
+    } as NestRequest;
+
+    const mockProvider = await fitnessController.getDatasourceForUser(
+      mockRequest,
+      { id: 'Mock' },
```

Comment:
Is `Mock` in lowercase intentional or should it be uppercase? Linked to my point of extracting constants 😉.

Reply:
The parameter is not used as the function is mocked to return a specific value. But undefined would work.